leshan-core/src/main/java/org/eclipse/leshan/senml/SenMLPackSerDes.java

```diff
@@ -126,4 +131,64 @@ public SenMLPack deserialize(JsonArray array) {
 
         return pack;
     }
+
+    public byte[] serializeToCbor(SenMLPack pack) {
+        CBORFactory factory = new CBORFactory();
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+        try {
+            CBORGenerator generator = factory.createGenerator(out);
+            generator.writeStartArray();
+
+            for (SenMLRecord record : pack.getRecords()) {
+                generator.writeStartObject();
+
+                if (record.getBaseName() != null && record.getBaseName().length() > 0) {
+                    generator.writeStringField("-2", record.getBaseName());
+                }
+
+                if (record.getBaseTime() != null) {
+                    generator.writeNumberField("-3", record.getBaseTime());
+                }
+
+                if (record.getName() != null && record.getName().length() > 0) {
+                    generator.writeStringField("0", record.getName());
+                }
+
+                if (record.getTime() != null) {
+                    generator.writeNumberField("6", record.getTime());
+                }
+
+                Type type = record.getType();
+                if (type != null) {
+                    switch (record.getType()) {
+                    case FLOAT:
+                        generator.writeNumberField("2", record.getFloatValue().floatValue());
+                        break;
+                    case BOOLEAN:
+                        generator.writeBooleanField("4", record.getBooleanValue());
+                        break;
+                    case OBJLNK:
+                        generator.writeStringField("vlo", record.getObjectLinkValue());
+                        break;
+                    case OPAQUE:
+                        generator.writeStringField("8", Base64.encodeBase64String(record.getOpaqueValue()));
+                    case STRING:
+                        generator.writeStringField("3", record.getStringValue());
+                        break;
+                    default:
+                        break;
+                    }
+                }
+                generator.writeEndObject();
+            }
+
+            generator.writeEndArray();
+            generator.close();
+        } catch (Exception e) {
+            throw new CodecException(e.getLocalizedMessage(), e);
```

Comment:
Instead of e.getLocalizedMessage() which is already available in e parameter. Should we add a more contextual  exception message, like : "Unable to serialize to Cbor." ?

Reply:
In addition, do you think CodecException can be used here, because of CodecException is thrown when a problem occurs during LWM2M node encoding or decoding. Or to extend a new Exception class for SenML-CBOR, like TlvException and LwM2MJsonException.

Reply:
Yep maybe this is even better. You can create a `SenMLCborException`.